src/main/java/dev/blaauwendraad/masker/json/KeyMatcher.java

```diff
@@ -173,6 +174,30 @@ public KeyMaskingConfig getMaskConfigIfMatched(byte[] bytes, int keyOffset, int
         }
     }
 
+    /**
+     * An overload of {@code dev.blaauwendraad.masker.json.KeyMatcher#getMaskConfigIfMatched(byte[], int, int, java.util.Iterator)} specifically for jsonpath look-ups.
+     */
+    @CheckForNull
+    public KeyMaskingConfig getMaskConfigIfMatched(byte[] bytes, Iterator<JsonPathSegmentReference> jsonPath) {
```

Comment:
I think it would be better to simply call the `getMaskConfigIfMatched(bytes, 0, 0, jsonPath)` from here. This basically ensures that return is called immediately

Reply:
-1? 😅

Reply:
I thought about it too! But it is "index out of bound" for "knownByteLengths" array. i could have made a special check for that, but decided that creating an overload is clearer.

Reply:
I have tried with additional check and I think it would be slightly better, mostly because `getMaskConfigIfMatched` contains quite a lot of comments on why it does things in certain way, e.g.
<img width="569" alt="image" src="https://github.com/Breus/json-masker/assets/15277543/ea80666d-c4b3-4021-80c0-3d4fcb85491e">

WDYT about [this](https://github.com/Breus/json-masker/files/14421650/Use_-1_for_key_length.patch)?

Reply:
I think you are right. This way we don't have to duplicate the logic and have only one place where we would need to update it if anything.

Reply:
Committed the fix